tests/layer_tests/tensorflow_tests/test_tf_ScatterND.py

```diff
@@ -73,3 +75,56 @@ def test_tf_scatter_nd(self, params, ie_device, precision, ir_version, temp_dir,
                                                                    use_legacy_frontend=use_legacy_frontend),
                    ie_device, precision, temp_dir=temp_dir, ir_version=ir_version,
                    use_legacy_frontend=use_legacy_frontend, **params)
+
+class TestTFScatterNDComplex(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        assert 'param_real:0' in inputs_info, "Test error: inputs_info must contain `param_real`"
+        assert 'param_imag:0' in inputs_info, "Test error: inputs_info must contain `param_imag`"
+        updates_shape = inputs_info['param_real:0']
+        inputs_data = {}
+        inputs_data['param_real:0'] = rng.integers(-10, 10, updates_shape).astype(np.float32)
+        inputs_data['param_imag:0'] = rng.integers(-10, 10, updates_shape).astype(np.float32)
+
+        return inputs_data
+
+    def create_tf_scatternd_complex_placeholder_const_net(self, x_shape, indices, updates_shape, ir_version,
+                                                  use_legacy_frontend):
+        import tensorflow as tf
+        tf.compat.v1.reset_default_graph()
+        with tf.compat.v1.Session() as sess:
+            param_real = tf.compat.v1.placeholder(tf.float32, updates_shape, 'param_real')
+            param_imag = tf.compat.v1.placeholder(tf.float32, updates_shape, 'param_imag')
+
+            tf_indices = tf.constant(indices)
+            tf_shape = tf.constant(x_shape)
+
+            complex = tf.raw_ops.Complex(real=param_real, imag=param_imag)
+
+            result = tf.scatter_nd(indices=tf_indices, updates=complex, shape=tf_shape, name="Operation")
+            tf.raw_ops.Real(input=result)
+            tf.raw_ops.Imag(input=result)
+
+            tf.compat.v1.global_variables_initializer()
+
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    test_data = [
+        dict(x_shape=[8], indices=[[4], [3], [1], [7]], updates_shape=[4]),
+        dict(x_shape=[10], indices=[[0], [2], [4], [6], [8]], updates_shape=[5]),
+        dict(x_shape=[5, 5], indices=[[0, 0], [1, 1], [2, 2], [3, 3]], updates_shape=[4]),
+        dict(x_shape=[3, 3, 3], indices=[[0, 0, 0], [1, 1, 1], [2, 2, 2]], updates_shape=[3]),
+    ]
+
```

Comment:
please add one more parameter `indices_type` that can be int16, int32, int64. `shape` is of the same type as `indices`

Reply:
@rkazants So, I did tried to add `indices_type` for types mentioned above but it seems like there is a problem with the int16 type.

```
try:
    c_op = pywrap_tf_session.TF_FinishOperation(op_desc)
except errors.InvalidArgumentError as e:
    # Convert to ValueError for backwards compatibility.
>   raise ValueError(e.message)

E   ValueError: Input tensor must be int32 or int64, but was int16

ov-py310/lib/python3.10/site-packages/tensorflow/python/framework/ops.py:1020: ValueError
```

Let me know if I should skip it then.

Reply:
if this TF error, problem looks in documentation or in TF implementation. So let us skip int16 then.